File: lib/layer2_core/light/e_error_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EErrorLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl41;
      case 'standard': return cstl41;
      case 'blue': return cbl41;
      case 'green': return cgl41;
      case 'red': return crl41;
      case 'nothing': return cnol41;
      default: return cstl41;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EErrorLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EErrorLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl41
            AppTheme.STANDARD -> R.color.cstl41
            AppTheme.BLUE -> R.color.cbl41
            AppTheme.GREEN -> R.color.cgl41
            AppTheme.RED -> R.color.crl41
            AppTheme.NOTHING -> R.color.cnol41
        }
    }
}
